### PR TITLE
introduce checker for  the result of nodeInfo.Node()

### PR DESCRIPTION
**Pull request by zouyee**
**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:
introduce checker for  the result of nodeInfo.Node()

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/nodelabel/node_label.go [hunk 1]
```diff
@@ -121,7 +124,7 @@ func (pl *NodeLabel) Filter(ctx context.Context, _ *framework.CycleState, pod *v
 // Score invoked at the score extension point.
 func (pl *NodeLabel) Score(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeName string) (int64, *framework.Status) {
 	nodeInfo, err := pl.handle.SnapshotSharedLister().NodeInfos().Get(nodeName)
-	if err != nil {
+	if err != nil || nodeInfo.Node() == nil {
```
Comment (wgliang) [inline pkg/scheduler/framework/plugins/nodelabel/node_label.go]: When `nodeInfo.Node () == nil`, whether it needs to be reflected in the log?

**Review by wgliang — APPROVED** [review]

File: pkg/scheduler/framework/plugins/nodelabel/node_label_test.go [hunk 2]
```diff
@@ -73,69 +73,87 @@ func TestValidateNodeLabelArgs(t *testing.T) {
 func TestNodeLabelFilter(t *testing.T) {
 	label := map[string]string{"foo": "any value", "bar": "any value"}
 	var pod *v1.Pod
+	node := v1.Node{ObjectMeta: metav1.ObjectMeta{Labels: label}}
 	tests := []struct {
-		name    string
-		rawArgs string
-		res     framework.Code
+		name      string
+		nodeExist bool
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/nodelabel/node_label_test.go]: I wouldn't complicate all the unit tests for this case. Just create a separate, much simpler test.

**Comment by ahg-g** [comment]
please change the type to cleanup because there is really no code path in the scheduler that could lead to this error.

File: pkg/scheduler/framework/plugins/nodelabel/node_label_test.go [hunk 3]
```diff
@@ -244,3 +244,56 @@ func TestNodeLabelScore(t *testing.T) {
...
+			if err != nil {
+				t.Fatalf("Failed to create plugin: %v", err)
+			}
+			status := p.(framework.FilterPlugin).Filter(context.TODO(), nil, pod, nodeInfo)
+			if status.Code() != test.res {
+				t.Errorf("Status mismatch. got: %v, want: %v", status.Code(), test.res)
+			}
+		})
+	}
+}
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/nodelabel/node_label_test.go]: it is just a single test, and I don't expect other tests to be added here, please simplify the whole test to.

```
t.Run("node does not exist", func(t *testing.T) {
			nodeInfo := schedulernodeinfo.NewNodeInfo()
			p, err := New(nil, nil)
			if err != nil {
				t.Fatalf("Failed to create plugin: %v", err)
			}
			status := p.(framework.FilterPlugin).Filter(context.TODO(), nil, pod, nodeInfo)
			if status.Code() != framework.Error {
				t.Errorf("Status mismatch. got: %v, want: %v", status.Code(), test.res)
			}
		})
```

File: pkg/scheduler/framework/plugins/nodelabel/node_label_test.go [hunk 4]
```diff
@@ -244,3 +244,56 @@ func TestNodeLabelScore(t *testing.T) {
...
+			p, err := New(nil, fh)
+			if err != nil {
+				t.Fatalf("Failed to create plugin: %+v", err)
+			}
+			_, status := p.(framework.ScorePlugin).Score(context.Background(), nil, nil, "")
+			if status.Code() != test.res {
+				t.Errorf("Wrong score. got %#v, want %#v", status, test.res)
+			}
+		})
+	}
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/nodelabel/node_label_test.go]: ditto, please simplify it to:

```
t.Run("node does not exist", func(t *testing.T) {
			fh, _ := framework.NewFramework(nil, nil, nil, framework.WithSnapshotSharedLister(nodeinfosnapshot.NewEmptySnapshot()))
			p, err := New(nil, fh)
			if err != nil {
				t.Fatalf("Failed to create plugin: %+v", err)
			}
			_, status := p.(framework.ScorePlugin).Score(context.Background(), nil, nil, "")
			if status.Code() != framework.Error {
				t.Errorf("Wrong score. got %#v, want %#v", status, test.res)
			}
		})
	}
```

File: pkg/scheduler/framework/plugins/nodelabel/node_label_test.go (hunk 3 above)
Comment (zouyee) [reply pkg/scheduler/framework/plugins/nodelabel/node_label_test.go]: Thanks, It has been done.

**Pull request merged**